controllers/topology/desired_state_test.go

```diff
@@ -419,6 +419,124 @@ func TestComputeCluster(t *testing.T) {
 	g.Expect(obj.Spec.ControlPlaneRef).To(Equal(objToRef(controlPlane)))
 }
 
+func TestComputeMachineDeployment(t *testing.T) {
+	workerInfrastructureMachineTemplate := newFakeInfrastructureMachineTemplate(metav1.NamespaceDefault, "linux-worker-inframachinetemplate").Obj()
+	workerBootstrapTemplate := newFakeBootstrapTemplate(metav1.NamespaceDefault, "linux-worker-bootstraptemplate").Obj()
+
+	labels := map[string]string{"fizz": "buzz", "foo": "bar"}
+	annotations := map[string]string{"annotation-1": "annotation-1-val"}
+
+	fakeClass := newFakeClusterClass(metav1.NamespaceDefault, "class1").
+		WithWorkerMachineDeploymentClass("linux-worker", labels, annotations, workerInfrastructureMachineTemplate, workerBootstrapTemplate).
+		Obj()
+	class := &clusterTopologyClass{
+		clusterClass: fakeClass,
+		machineDeployments: map[string]machineDeploymentTopologyClass{
+			"linux-worker": {
+				metadata: clusterv1.ObjectMeta{
+					Labels:      labels,
+					Annotations: annotations,
+				},
+				bootstrapTemplate:             workerBootstrapTemplate,
+				infrastructureMachineTemplate: workerInfrastructureMachineTemplate,
+			},
+		},
+	}
+
+	current := &clusterTopologyState{
+		cluster: newFakeCluster(metav1.NamespaceDefault, "cluster1").Obj(),
+	}
+
+	replicas := 5
+	mdTopology := clusterv1.MachineDeploymentTopology{
+		Metadata: clusterv1.ObjectMeta{
+			Labels: map[string]string{"foo": "baz"},
+		},
+		Class:    "linux-worker",
+		Name:     "big-pool-of-machines",
+		Replicas: &replicas,
+	}
+
+	t.Run("Generates the machine deployment and the referenced templates", func(t *testing.T) {
+		g := NewWithT(t)
+		actual, err := computeMachineDeployment(class, current, mdTopology)
+		g.Expect(err).ToNot(HaveOccurred())
+
+		actualMd := actual.object
+		g.Expect(*actualMd.Spec.Replicas).To(Equal(int32(5)))
```

Comment:
```suggestion
		g.Expect(*actualMd.Spec.Replicas).To(Equal(int32(replicas)))
```
nit (in case you prefer it that way)